opacclient/libopac/src/main/java/de/geeksfactory/opacclient/apis/Wise.kt

```diff
@@ -0,0 +1,563 @@
+package de.geeksfactory.opacclient.apis
+
+import de.geeksfactory.opacclient.i18n.StringProvider
+import de.geeksfactory.opacclient.networking.HttpClientFactory
+import de.geeksfactory.opacclient.networking.NotReachableException
+import de.geeksfactory.opacclient.objects.*
+import de.geeksfactory.opacclient.searchfields.BarcodeSearchField
+import de.geeksfactory.opacclient.searchfields.DropdownSearchField
+import de.geeksfactory.opacclient.searchfields.SearchField
+import de.geeksfactory.opacclient.searchfields.SearchQuery
+import de.geeksfactory.opacclient.utils.jsonObject
+import okhttp3.Interceptor
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.util.*
+import javax.crypto.Mac
+import javax.crypto.spec.SecretKeySpec
+import kotlin.experimental.and
+import kotlin.math.floor
+
+data class Authentication(val libraryId: String, val patronSystemId: String,val token: String, val membershipExpiryDate: org.joda.time.LocalDate )
+
+lateinit var apiKeyId: String
+lateinit var apiKey: String
+lateinit var applicationName: String
+var auth: Authentication? =  null
+
+/**
+ * OpacApi implementation for Wise owned by OCLC https://www.oclc.org/en/wise.html
+ *
+ *
+ * @author Henk Klijn Hesselink, April 2021
+ */
+
+
+class WiseKeyInterceptor : Interceptor {
```

Comment:
Hmm, I'm not completely sure if an interceptor is the best design to add an Authorization header. Is this a best practise with okhttp? It feels like it makes it harder to follow the code and debug what is actually being sent.

I'd personally prefer extending OkhttpBaseApi with something like

```
open fun prepareRequestBuilder(): RequestBilder
```

that can then be used to add the header to every request.